qiskit_experiments/curve_analysis/utils.py

```diff
@@ -222,6 +224,166 @@ def eval_with_uncertainties(
     return wrapfunc(x=x, **sub_params)
 
 
+def shot_weighted_average(
```

Comment:
These functions are tied to ScatterTable column structure. I cannot reuse existing ones.